client/my-sites/email/email-non-domain-owner-message/index.tsx

```diff
@@ -0,0 +1,139 @@
+import { stringify } from 'qs';
+import { useDispatch } from 'react-redux';
+import { useDomainOwnerUserName } from 'calypso/components/data/query-domain-owner-username';
+import PromoCard from 'calypso/components/promo-section/promo-card';
+import { ResponseDomain } from 'calypso/lib/domains/types';
+import { emailManagementPurchaseNewEmailAccount } from 'calypso/my-sites/email/paths';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+
+import './style.scss';
+
+type EmailNonDomainOwnerMessageProps = {
+	domain?: ResponseDomain;
+	selectedSite?: SiteDetails | null;
+	usePromoCard: boolean;
+	source: 'email-comparison' | 'email-management';
+};
+
+export const EmailNonDomainOwnerMessage = ( props: EmailNonDomainOwnerMessageProps ) => {
+	const { domain, selectedSite, source, usePromoCard } = props;
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const ownerUserName = useDomainOwnerUserName( selectedSite, domain );
+
+	const isPrivacyAvailable = domain?.privacyAvailable;
+
+	const buildQueryString = ( parameters = {} ) =>
+		parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = encodeURIComponent(
+			emailManagementPurchaseNewEmailAccount(
+				selectedSite?.slug ?? '',
+				domain?.name ?? '',
+				null,
+				'login-redirect'
+			)
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const contactOwnerUrl = `https://privatewho.is/${ buildQueryString( {
+		s: domain?.name,
+	} ) }`;
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'owner_contact' | 'user_login' | 'contact_support' ) => {
+		const properties = {
+			owner_login: ownerUserName,
+		};
+
+		if ( eventType === 'owner_contact' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_owner_contact_click`, properties ) );
+
+			return;
+		} else if ( eventType === 'contact_support' ) {
+			dispatch( recordTracksEvent( `calypso_email_providers_contact_support_click`, properties ) );
+
+			return;
+		}
+
+		dispatch( recordTracksEvent( `calypso_email_providers_user_login_click`, properties ) );
+	};
+
+	const translateOptions = {
+		components: {
+			loginLink: (
+				<a
+					rel="noopener noreferrer"
+					href={ loginUrl }
+					onClick={ () => onClickLink( 'user_login' ) }
+				/>
+			),
+			reachOutLink: isPrivacyAvailable ? (
+				<a
+					href={ contactOwnerUrl }
+					onClick={ () => onClickLink( 'owner_contact' ) }
+					rel="noopener noreferrer"
+					target="_blank"
+				/>
+			) : (
+				<></>
+			),
+			contactSupportLink: (
+				<a
+					href="https://wordpress.com/help/contact"
+					onClick={ () => onClickLink( 'contact_support' ) }
+					rel="noopener noreferrer"
+					target="_blank"
+				/>
+			),
+		},
+		args: {
+			ownerUserName,
+			selectedDomainName: domain?.name,
+		},
+	};
+
+	function getReason() {
```

Comment:
I initially thought of this as a potential render optimization (using an inline function in a React component makes it difficult for React to avoid unnecessary rerenders). However, after thinking about it, I'm not 100% sure that it actually has an impact in this case.

So this is potentially a subjective issue, although I still think it would be clearer. 

My thinking is that we ought to remove the function and instead just use a regular `if` statement to assign a value to a variable. Like this:

```tsx
let text: TranslateResult;

if ( source === 'email-comparison' ) {
    text = translate( ... );
} else if ( source === 'email-management' ) {
    text = translate( ... );
}
```